src/Analysis/Ast/Impl/Specializations/Typing/Types/GenericTypeParameter.cs

```diff
@@ -27,7 +27,6 @@ namespace Microsoft.Python.Analysis.Specializations.Typing.Types {
     internal sealed class GenericTypeParameter : PythonType, IGenericTypeParameter {
         public GenericTypeParameter(
             string name,
-            IPythonModule declaringModule,
             IReadOnlyList<IPythonType> constraints,
             object bound,
             object covariant,
```

Comment:
same with contravariant

Reply:
It would make them `IPythonConstant` I guess...

Reply:
Actually, it can come as anything. I.e. nothing prevents user from typing `bound=3, covariant=7`. It is a good question should we report diagnostics on this, but from this object perspective parameter has value.

Reply:
But if we store bound=3, covariant=7, then what can we do with that? I would say it's better to just leave them as null because bound=3 is the same as unknown, and having covariant as 7 also is unknown because it's supposed to be a boolean.

Reply:
No, we should store exactly as it was or else we can't test that restored module is identical to the source.

Reply:
If that's the case, then we should convert constraints into a list of objects.